tests/unit/test_vcs.py

```diff
@@ -110,3 +111,29 @@ def test_bazaar_simple_urls():
     assert launchpad_bzr_repo.get_url_rev() == (
         'lp:MyLaunchpadProject', None,
     )
+
+
+def test_subversion_remove_auth_from_url():
```

Comment:
This is perfect for parametrized tests https://github.com/pypa/pip/blob/develop/tests/unit/test_vcs.py#L53-L58